### PR TITLE
Corrects wrong insertion point for 'await' at uncovered 'async' site.

### PR DESCRIPTION
**Pull request by Rajveer100**
Fixes Issue #65913

### CODE REVIEW

**Comment by Rajveer100** [comment]
@YOCKOW @AnthonyLatsis

**Review by AnthonyLatsis — CHANGES_REQUESTED** [review]
Please add a test. As is, this change will most likely break some tests, so I’m going to trigger CI for you to inspect the failures.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test macOS

**Comment by Rajveer100** [comment]
> Please add a test. As is, this change will most likely break some tests, so I’m going to trigger CI for you to inspect the failures.

Well, I do suspect that too as we must probably handle the cases separately for each type of expression in an async await call expression (ex. `tryExpr`, `closure`, `_`, etc), let's see!

Regarding tests, I presume there would be cases as mentioned above, let me know if I missed out any!

File: lib/Sema/TypeCheckEffects.cpp [hunk 1]
```diff
@@ -2867,8 +2867,13 @@ class CheckEffectsCoverage : public EffectsHandlingWalker<CheckEffectsCoverage>
...
 
-    Ctx.Diags.diagnose(anchor->getStartLoc(), diag::async_expr_without_await)
-      .fixItInsert(awaitInsertLoc, "await ")
+    auto anchorDecl = dyn_cast<DeclRefExpr>(anchor);
+    auto anchorVar = dyn_cast<VarDecl>(anchorDecl->getDecl());
+    auto anchorIdentifier = anchorVar->getName().str().str();
+    auto awaitInsertString = anchorIdentifier + " = await ";
+    
+    Ctx.Diags.diagnose(awaitInsertLoc, diag::async_expr_without_await)
+      .fixItInsert(awaitInsertLoc, StringRef(awaitInsertString))
```
[the remark this answers is not in the document]
Comment (Rajveer100) [reply lib/Sema/TypeCheckEffects.cpp:2876]: If you meant a normal variable declaration (i.e. `var _ = asyncFunc()`, we would get the following error:

```
error: 'async' call in a function that does not support concurrency
```

It actually doesn't provide a fixit to add the `await` keyword.

**Comment by Rajveer100** [comment]
After some debugging I found that only these two lines by themselves are causing an issue:

`auto anchorVar = dyn_cast<VarDecl>(anchorDecl->getDecl());`

Regarding the tests itself, before they are even checked, there seems to be internal error (probably memory error/bad access/similar), hence shows 'Please submit a bug report' as `stderr`.

**Comment by AnthonyLatsis** [comment]
For the future, don’t be shy to ping  someone once every 2 weeks or so in PRs if you want a review. We sometimes *do* forget to get back to stuff.

**Comment by Rajveer100** [comment]
> For the future: Don’t be shy to ping someone once every 2 weeks or so in PRs if you want a review. We sometimes _do_ forget to get back to stuff.

Indeed! 

PS: By the way I am excited for GSoC 2024, hoping to get land proposals just like last year and getting accepted this time!

**Pull request closed without merging**